modules/dnn/include/opencv2/dnn/shape_utils.hpp

```diff
@@ -240,9 +240,9 @@ static inline std::ostream& operator<<(std::ostream &out, const std::vector<_Tp>
 static inline
 int normalize_axis(int axis, int dims)
 {
-    CV_Check(axis, axis >= -dims && axis < dims, "");
-    axis = (axis < 0) ? (dims + axis) : axis;
-    CV_DbgCheck(axis, axis >= 0 && axis < dims, "");
+    CV_Assert(dims >= 0);
+    CV_Check(axis, axis >= -dims && axis <= dims, "");
```

Comment:
> `axis <= dims`

We should not allow `axis == dims`.
It is not allowed in classic index/stride rules (e.g. in Python)

```
>>> a = [1, 2, 3]
>>> a[-3]
1
>>> a[-2]
2
>>> a[-1]
3
>>> a[0]
1
>>> a[1]
2
>>> a[2]
3
>>> a[3]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
>>>
```